paimon-core/src/main/java/org/apache/paimon/catalog/AbstractCatalog.java

```diff
@@ -326,7 +337,7 @@ protected abstract TableSchema getDataTableSchema(Identifier identifier)
 
     @VisibleForTesting
     public Path getDataTableLocation(Identifier identifier) {
-        return newTableLocation(warehouse(), identifier);
+        return newTableLocation(newDatabasePath(identifier.getDatabaseName()), identifier);
```

Comment:
I'm a bit confused about how much code we need to modify. Should we just handle it directly here.

Just: `new Path(databasePath, tableName)` here?